lib/dependency-manager-adapters/npm.js

```diff
@@ -7,13 +7,37 @@ var path = require('path');
 var extend = require('extend');
 var debug = require('debug')('ember-try:dependency-manager-adapter:npm');
 var rimraf = RSVP.denodeify(require('rimraf'));
+var execFileSync = require('child_process').execFileSync;
 
 module.exports = CoreObject.extend({
   init: function() {
     this._super.apply(this, arguments);
     this.run = this.run || require('../utils/run');
+    this.adapterInit();
   },
-  configKey: 'npm',
+  adapterInit() {
+    this.yarnLockPresent = fs.existsSync(path.join(this.cwd, this.yarnLock));
+    let yarnManagerPresent;
+    try {
+      execFileSync('yarn', [].concat(['--version']));
```

Comment:
`init` code is intended to run in sync. This behaviour is expected by i.e. `tasks/try-each.js` and probably by some tests and other code as well. This piece is ran in `init` because it seems the single input point for adapter. 

To make it async we can either run it in both `cleanup` and `setup` as those are async functions, but then a little bit more sophisticated logic would be required to avoid double execution or I can try to mitigate `sync` requirement for `init` but that would require works on both `bower` `dependency-manager-adapter` and probably `tasks/try-each.js` plus tests code.